Lib/_ios_support.py

```
@@ -0,0 +1,70 @@
+import sys
+try:
+    from ctypes import cdll, c_void_p, c_char_p, util
+except ImportError:
+    # ctypes is an optional module. If it's not present, we're limited in what
+    # we can tell about the system, but we don't want to prevent the platform
+    # module from working.
+    objc = None
+else:
+    # ctypes is available. Load the ObjC library, and wrap the objc_getClass,
+    # sel_registerName and objc_msgSend methods
+    objc = cdll.LoadLibrary(util.find_library("objc"))
+    if objc._name:
```

We should tolerate ctypes being unavailable, but if a fundamental iOS system library is apparently missing, then something has gone badly wrong and we should probably throw an exception rather than allowing the problem to affect higher-level code where its cause will be harder to understand.

Also, it would be clearer to test the return value of find_library before passing it to LoadLibrary, especially since the meaning of `LoadLibrary(None)` has never been clearly documented (#78773).

---

Agreed; I've raised a RuntimeError if ObjC can't be loaded.